gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -308,6 +315,40 @@ var gpii = fluid.registerNamespace("gpii");
         });
     };
 
+    /** Upgrades a promise rejection payload (or Error) by suffixing an additional "while" reason into its "message" field
+     * @param originError {Object|Error} A rejection payload. This should (at least) have the member `isError: true` set, as well as a String `message` holding a rejection reason.
+     * @param whileMsg {String} A message describing the activity which led to this error
+     * @return {Object} The rejected payload formed by shallow cloning the supplied argument (if it is not an `Error`) and suffixing its `message` member
```

Comment:
Looking at the code, I understand what this return description means. But before reading the code, I had a question: what happens if it is an Error? Maybe could rework the wording or structure to make it clear that the message is always updated, either in-place (Error), or on a copy (Object).